### PR TITLE
mds: inode_t flags may not be protected by the policylock during set_vxattr

### PR DESCRIPTION
**Pull request by batrick**
## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "quincy"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

- When filling out the below checklist, you may click boxes directly in the GitHub web UI.  When entering or editing the entire PR message in the GitHub web UI editor, you may also select a checklist item by adding an `x` between the brackets: `[x]`.  Spaces and capitalization matter when checking off items this way.

## Checklist
- Tracker (select at least one)
  - [x] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [x] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
- `jenkins test rook e2e`
</details>

### CODE REVIEW

**Comment by mchangir** [comment]
I wonder if this was the cause of missing `ceph.dir.subvolume` in some upstream and downstream cases.

File: src/mds/MDSDaemon.cc [hunk 1]
```diff
@@ -364,11 +364,11 @@ void MDSDaemon::set_up_admin_socket()
                                      asok_hook,
                                      "show cache status");
   ceph_assert(r == 0);
-  r = admin_socket->register_command("quiesce path"
+  r = admin_socket->register_command("lock path"
```
Comment (batrick) [inline src/mds/MDSDaemon.cc:367]: whoops! Didn't mean to do this.

**Comment by batrick** [comment]
> I wonder if this was the cause of missing `ceph.dir.subvolume` in some upstream and downstream cases.

The inode still gets projected but ... maybe?

**Comment by batrick** [comment]
This PR is under test in [https://tracker.ceph.com/issues/65530](https://tracker.ceph.com/issues/65530).

**Comment by batrick** [comment]
jenkins test api

File: src/mds/MDSRank.cc [hunk 2]
```diff
@@ -3534,6 +3536,30 @@ int MDSRank::command_quiesce_path(Formatter* f, const cmdmap_t& cmdmap, std::ost
...
+    if (!got) {
+      ss << "missing path";
+      return -CEPHFS_EINVAL;
+    }
+  }
+
+  std::vector<std::string> locks;
+  cmd_getval(cmdmap, "locks", locks);
+
+  f->open_object_section("quiesce");
```
Comment (leonid-s-usov) [inline src/mds/MDSRank.cc:3553]: copy-paste? This is not seen in the render because it's the root object, but still, should probably be `lock`?

**Review by leonid-s-usov — APPROVED** [review]
Wow, that's a good catch!

Please update the name of this PR because it's not about replication but locking.

On a separate note: love the `lock path` admin command, very much needed!

File: src/mds/MDSRank.cc (hunk 2 above)
Comment (batrick) [reply src/mds/MDSRank.cc]: fixed

**Comment by batrick** [comment]
jenkins test api

**Comment by batrick** [comment]
This PR is under test in [https://tracker.ceph.com/issues/65562](https://tracker.ceph.com/issues/65562).

**Comment by batrick** [comment]
This PR is under test in [https://tracker.ceph.com/issues/65596](https://tracker.ceph.com/issues/65596).

**Comment by batrick** [comment]
jenkins test make check

**Comment by batrick** [comment]
https://tracker.ceph.com/projects/cephfs/wiki/Main#2024-04-20

File: qa/suites/fs/volumes/tasks/volumes/test/basic.yaml [hunk 3]
```diff
@@ -5,4 +5,4 @@ tasks:
         - tasks.cephfs.test_volumes.TestVolumes
         - tasks.cephfs.test_volumes.TestSubvolumeGroups
         - tasks.cephfs.test_volumes.TestSubvolumes
-        - tasks.cephfs.test_subvolume.TestSubvolume
+        - tasks.cephfs.test_subvolume
```
Comment (batrick) [inline qa/suites/fs/volumes/tasks/volumes/test/basic.yaml:8]: Needs rerun because this was missing.

**Comment by batrick** [comment]
jenkins test make check arm64

**Comment by batrick** [comment]
This PR is under test in [https://tracker.ceph.com/issues/65661](https://tracker.ceph.com/issues/65661).

**Comment by batrick** [comment]
jenkins test make check arm64

**Comment by batrick** [comment]
This PR is under test in [https://tracker.ceph.com/issues/65694](https://tracker.ceph.com/issues/65694).

**Comment by batrick** [comment]
https://tracker.ceph.com/issues/65694#note-2

**Pull request merged**